File: frontend/src/types/order.ts
export type OrderStatus = "new" | "selection" | "pickup" | "delivery" | "closed" | "cancelled";
export type TaskStatus = "pending" | "in_progress" | "waiting_client" | "approved" | "rejected" | "completed" | "cancelled";

export interface Photo {
  id: number;
  file_url: string;
  created_at: string;
}

export interface SelectionTask {
  id: number;
  order_id: number;
  picker_id: number | null;
  status: TaskStatus;
  note: string | null;
  photos: Photo[];
  created_at: string;
  updated_at: string;
}

export interface PickupTask {
  id: number;
  order_id: number;
  picker_id: number | null;
  seller_address: string | null;
  seller_lat: number | null;
  seller_lng: number | null;
  needs_inspection: boolean;
  status: TaskStatus;
  note: string | null;
  photos: Photo[];
  created_at: string;
  updated_at: string;
}

export interface DeliveryTask {
  id: number;
  order_id: number;
  picker_id: number | null;
  delivery_address: string | null;
  is_third_party_carrier: boolean;
  status: TaskStatus;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface Order {
  id: number;
  client_id: number;
  drom_url: string | null;
  description: string | null;
  target_price: number | null;
  comment: string | null;
  prepaid_to_seller: boolean;
  status: OrderStatus;
  selection_task: SelectionTask | null;
  pickup_task: PickupTask | null;
  delivery_task: DeliveryTask | null;
  photos: Photo[];
  created_at: string;
  updated_at: string;
}

export interface OrderListItem {
  id: number;
  drom_url: string | null;
  description: string | null;
  status: OrderStatus;
  created_at: string;
}

export interface OrderCreatePayload {
  drom_url?: string | null;
  description?: string | null;
  target_price?: number | null;
  comment?: string | null;
  prepaid_to_seller: boolean;
}

File: backend/app/schemas/order.py
from datetime import datetime
from pydantic import BaseModel, Field, model_validator
from app.models.order import OrderStatus, TaskStatus


class OrderCreate(BaseModel):
    drom_url: str | None = Field(None, max_length=2048)
    description: str | None = Field(None, max_length=5000)
    target_price: float | None = Field(None, ge=0)
    comment: str | None = Field(None, max_length=2000)
    prepaid_to_seller: bool = False

    @model_validator(mode="after")
    def require_drom_or_description(self):
        if not self.drom_url and not self.description:
            raise ValueError("Укажите ссылку Drom или описание детали")
        return self


class OrderUpdate(BaseModel):
    status: OrderStatus | None = None
    comment: str | None = None


class PhotoResponse(BaseModel):
    id: int
    file_url: str
    created_at: datetime

    model_config = {"from_attributes": True}


class TaskBrief(BaseModel):
    id: int
    status: TaskStatus

    model_config = {"from_attributes": True}


class SelectionTaskResponse(BaseModel):
    id: int
    order_id: int
    picker_id: int | None
    status: TaskStatus
    note: str | None
    photos: list[PhotoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PickupTaskResponse(BaseModel):
    id: int
    order_id: int
    picker_id: int | None
    seller_address: str | None
    seller_lat: float | None
    seller_lng: float | None
    needs_inspection: bool
    status: TaskStatus
    note: str | None
    photos: list[PhotoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DeliveryTaskResponse(BaseModel):
    id: int
    order_id: int
    picker_id: int | None
    delivery_address: str | None
    is_third_party_carrier: bool
    status: TaskStatus
    note: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: int
    client_id: int
    drom_url: str | None
    description: str | None
    target_price: float | None
    comment: str | None
    prepaid_to_seller: bool
    status: OrderStatus
    selection_task: SelectionTaskResponse | None = None
    pickup_task: PickupTaskResponse | None = None
    delivery_task: DeliveryTaskResponse | None = None
    photos: list[PhotoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class OrderListItem(BaseModel):
    id: int
    drom_url: str | None
    description: str | None
    status: OrderStatus
    created_at: datetime

    model_config = {"from_attributes": True}


class TaskUpdate(BaseModel):
    status: TaskStatus | None = None
    note: str | None = None


class PickupTaskUpdate(TaskUpdate):
    seller_address: str | None = None
    seller_lat: float | None = None
    seller_lng: float | None = None


class DeliveryTaskUpdate(TaskUpdate):
    delivery_address: str | None = None
    is_third_party_carrier: bool | None = None
